PowerPointLabs/PowerPointLabs/Ribbon1.cs

```diff
@@ -49,32 +49,56 @@ public class Ribbon1 : Office.IRibbonExtensibility
         private ImageHandlerFactory ImageHandlerFactory { get; set; }
 
         private SupertipHandlerFactory SupertipHandlerFactory { get; set; }
+
+        private ContentHandlerFactory ContentHandlerFactory { get; set; }
+
+        private PressedHandlerFactory PressedHandlerFactory { get; set; }
+
+        private CheckBoxActionHandlerFactory CheckBoxActionHandlerFactory { get; set; }
         #endregion
 
         #region Action Framework entry point
 
         public void OnAction(Office.IRibbonControl control)
         {
-            var actionHandler = ActionHandlerFactory.CreateInstance(control.Id);
-            actionHandler.Execute(control.Id);
+            var actionHandler = ActionHandlerFactory.CreateInstance(control.Id, control.Tag);

```

Comment:
From Microsoft: If this attribute is omitted, the control's tag value SHOULD default to an empty string.